sdk/ml/azure-ai-ml/tests/internal/e2etests/test_pipeline_job.py

```diff
@@ -94,6 +109,47 @@ def pipeline_func():
             json.dumps(node_rest_dict, indent=2), json.dumps(mismatched_runsettings, indent=2)
         )
 
+    @pytest.mark.parametrize(
+        "test_case",
+        PARAMETERS_TO_TEST_WITH_OUTPUT,
+    )
+    @pytest.mark.disable_mock_code_hash
+    def test_register_output_for_anonymous_internal_component(
+        self,
+        client: MLClient,
+        test_case: tuple,
+    ):
+        (
+            yaml_path,
+            inputs,
+            runsettings_dict,
+            pipeline_runsettings_dict,
+            output_port_name,
+            output_name,
+            output_version,
+        ) = test_case
+        node_func: InternalComponent = load_component(yaml_path)
+
+        @pipeline()
+        def pipeline_func():
+            node = node_func(**inputs)
+            node.outputs[output_port_name].type = "uri_file"  # use this, in case that the type is path
```

Comment:
Why do we need to set this?

Reply:
I find that if we don't set type here, we will meet an error: `marshmallow.exceptions.ValidationError: Value 'path' passed is not in set ['uri_file', 'uri_folder', 'custom_model', 'mlflow_model', 'mltable', 'triton_model']`. As shown in the internal cocmponent yaml, this error is because the type of internal component output is `path`. I think there may be some gap between v2 component definition and internal component definition, therefore I set type here. 
![image](https://user-images.githubusercontent.com/49266998/221084057-d37c7cc5-8904-4335-9ddc-81b3ba1077a7.png)